From d4f5629fa3b52951c53ff6c62c082615b83bc5cd Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Wed, 23 Jul 2014 13:27:05 -0400
Subject: [input] synaptics: add min/max quirk for ThinkPad Edge E431

Message-id: <1406122030-5632-10-git-send-email-benjamin.tissoires@redhat.com>
Patchwork-id: 86348
O-Subject: [RHEL7.1 / RHEL7.0.z PATCH 09/14] Input: synaptics - add min/max quirk for ThinkPad Edge E431
Bugzilla: 1093449
RH-Acked-by: Aristeu S. Rozanski F. <arozansk@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Peter Hutterer <peter.hutterer@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1093449
Upstream: since v3.15, backported in stable
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7722437
Test: tested by myself on a t440s and a t410, was extensively tested in RHEL 6.6

commit 27a38856a948c3e8de30dc71647ff9e1778c99fc
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Wed Apr 23 13:02:35 2014 -0700

    Input: synaptics - add min/max quirk for ThinkPad Edge E431

    Cc: stable@vger.kernel.org
    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/input/mouse/synaptics.c b/drivers/input/mouse/synaptics.c
index ef9f491..d68d33f 100644
--- a/drivers/input/mouse/synaptics.c
+++ b/drivers/input/mouse/synaptics.c
@@ -1566,6 +1566,14 @@ static const struct dmi_system_id min_max_dmi_table[] __initconst = {
   .driver_data = (int []){1232, 5710, 1156, 4696},
  },
  {
+  /* Lenovo ThinkPad Edge E431 */
+  .matches = {
+   DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+   DMI_MATCH(DMI_PRODUCT_VERSION, "ThinkPad Edge E431"),
+  },
+  .driver_data = (int []){1024, 5022, 2508, 4832},
+ },
+ {
   /* Lenovo ThinkPad T431s */
   .matches = {
    DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
-- 
1.7.1